File: project/user/web/src/api/exportTemplate.js
import { api } from '@/api/auth'

/**
 * 查询启用的导出模板列表。
 * @returns {Promise<Array<{templateKey:string, name:string, platform:string, description:string, bgColor:string, textColor:string, visualStyle:Object, signatureText:string, signaturePosition:string, sortOrder:number}>>}
 */
export function listExportTemplates() {
  return api.get('/export-templates').then((res) => res.data || [])
}

File: project/user/api/src/main/java/com/aichuangzuo/user/modules/exporttemplate/entity/ExportTemplate.java
package com.aichuangzuo.user.modules.exporttemplate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@TableName("a_export_template")
public class ExportTemplate {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String templateKey;
    private String name;
    private String platform;
    private String description;
    private String bgColor;
    private String textColor;
    private String visualStyleJson;
    private String signatureText;
    private String signaturePosition;
    private Integer sortOrder;
    private Integer status;

    @TableLogic
    private Integer isDeleted;
}
